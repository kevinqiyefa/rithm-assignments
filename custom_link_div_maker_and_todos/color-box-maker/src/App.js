import React, { Component } from 'react';

import './App.css';
import BoxList from './BoxList';
import NewBoxForm from './NewBoxForm';

class App extends Component {
  state = { boxes: [] };

  handleCreateBox = newBox => {
    this.setState({ boxes: this.state.boxes.concat(newBox) });
  };

  handleRemoveBox = id => {
    this.setState({ boxes: this.state.boxes.filter(b => b.id !== id) });
  };

  render() {
    return (
      <div className="App">
        <BoxList boxes={this.state.boxes} deleteBox={this.handleRemoveBox} />
        <NewBoxForm createBox={this.handleCreateBox} />
      </div>
    );
  }
}

export default App;
