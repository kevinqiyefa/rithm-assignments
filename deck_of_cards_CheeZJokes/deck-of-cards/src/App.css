.App {
  text-align: center;
  font-size: 1.3em;
  margin-top: 5em;
}

.animate {
  transition: all 0.1s;
  -webkit-transition: all 0.1s;
}

.action-button {
  position: relative;
  padding: 10px 40px;
  margin: 0px 10px 10px 0px;
  border-radius: 10px;
  font-size: 25px;
  color: #fff;
  text-decoration: none;
  cursor: pointer;
}

.blue {
  font-size: 2em;
  margin-top: 3em;
  background-color: #3498db;
  border-bottom: 5px solid #2980b9;
  text-shadow: 0px -2px #2980b9;
}

ul {
  margin-top: 5em;
}
