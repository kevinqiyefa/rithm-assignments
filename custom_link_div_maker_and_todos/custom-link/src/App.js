import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import CustomLink from './CustomLink';
class App extends Component {
  state = { isDisable: false, target: '_self' };

  toggleDisableLink = () => {
    this.setState({ isDisable: !this.state.isDisable });
  };

  openLink = e => {
    if (this.state.isDisable) {
      e.preventDefault();
    }
    this.setState({ target: '_blank' });
  };

  render() {
    const links = [
      { href: 'https://www.rithmschool.com', text: 'Rithm School' },
      { href: 'https://facebook.github.io/react/', text: 'React Docs' },
      { href: 'https://www.codewars.com/dashboard', text: 'CodeWars' }
    ];

    return (
      <div className="App">
        {links.map((m, index) => (
          <CustomLink
            href={m.href}
            text={m.text}
            key={index}
            openLink={this.openLink}
            target={this.state.target}
          />
        ))}
        <button onClick={this.toggleDisableLink}> Disable Link</button>
      </div>
    );
  }
}

export default App;
