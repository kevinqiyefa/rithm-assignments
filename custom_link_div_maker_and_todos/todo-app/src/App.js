import React, { Component } from 'react';
import './App.css';
import NewTodoForm from './NewTodoForm';
import TodoList from './TodoList';

class App extends Component {
  state = { todoList: [] };

  createTask = task => {
    this.setState({ todoList: this.state.todoList.concat(task) });
  };

  removeTask = id => {
    this.setState({
      todoList: this.state.todoList.filter(task => task.id !== id)
    });
  };

  toggleEditing = (id, isEditing) => {
    this.setState({
      todoList: this.state.todoList.map(t => {
        if (t.id === id) t.isEditing = isEditing;
        return t;
      })
    });
  };

  toggleComplete = (id, isCheck) => {
    this.setState({
      todoList: this.state.todoList.map(t => {
        if (t.id === id) t.isComplete = isCheck;
        return t;
      })
    });
  };

  editTask = (id, task) => {
    this.setState({
      todoList: this.state.todoList.map(t => {
        if (t.id === id) t.task = task;
        return t;
      })
    });
  };

  render() {
    return (
      <div className="App">
        <NewTodoForm createTask={this.createTask} />
        <TodoList
          todoList={this.state.todoList}
          removeTask={this.removeTask}
          toggleEditing={this.toggleEditing}
          editTask={this.editTask}
          toggleChecked={this.toggleComplete}
        />
      </div>
    );
  }
}

export default App;
