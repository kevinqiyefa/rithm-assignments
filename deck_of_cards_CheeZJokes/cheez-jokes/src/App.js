import React, { Component } from 'react';
import axios from 'axios';
import JokeList from './JokeList';
import uuidv1 from 'uuid/v1';
import './App.css';

class App extends Component {
  state = {
    jokes: [],
    message: 'Loading....'
  };

  async componentDidMount() {
    try {
      let list = [];
      let unique = new Set();

      while (unique.size < 20) {
        const response = await axios.get(`https://icanhazdadjoke.com/`, {
          headers: { Accept: 'application/json' }
        });
        unique.add(response.data.joke);
      }
      [...unique].forEach(j => {
        list.push({ id: uuidv1(), joke: j, upvote: 0, downvote: 0 });
      });

      this.setState({
        jokes: list,
        message: ''
      });
    } catch (e) {
      alert('Opps somthing went wrong!');
    }
  }

  refreshJokes = () => {
    this.setState({
      jokes: [],
      message: 'Loading....'
    });
    this.componentDidMount();
  };

  addUpVote = id => {
    this.setState({
      jokes: this.state.jokes.map(j => {
        if (id === j.id) ++j.upvote;
        return j;
      })
    });
  };
  addDownVote = id => {
    this.setState({
      jokes: this.state.jokes.map(j => {
        if (id === j.id) ++j.downvote;
        return j;
      })
    });
  };

  render() {
    return (
      <div className="App">
        {!this.state.message ? (
          <div>
            <div className="App-Button">
              <button className="refresh" onClick={this.refreshJokes}>
                Click Me to Refresh!
              </button>
            </div>
            <JokeList
              jokes_list={this.state.jokes}
              upVote={this.addUpVote}
              downVote={this.addDownVote}
            />
          </div>
        ) : (
          <p>{this.state.message}</p>
        )}
      </div>
    );
  }
}

export default App;
