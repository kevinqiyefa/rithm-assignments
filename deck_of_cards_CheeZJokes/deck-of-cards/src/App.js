import React, { Component } from 'react';
import axios from 'axios';
import Card from './Card';
import './App.css';

class App extends Component {
  state = {
    images: [],
    showimages: []
  };
  displayCards = () => {
    let cards = this.state.images.slice();
    let randomimage = cards.pop();

    this.setState({
      images: cards,
      showimages: [...this.state.showimages, randomimage]
    });
  };
  async componentDidMount() {
    try {
      let img = [];
      const response = await axios.get(
        'https://deckofcardsapi.com/api/deck/new/draw/?count=52'
      );

      response.data.cards.forEach(card => img.push(card.image));

      this.setState({
        images: img
      });
    } catch (e) {
      alert('Opps, something went wrong');
    }
  }
  render() {
    const cards = this.state.showimages.map((i, idx) => (
      <Card key={idx} src={i} />
    ));
    return (
      <div className="App">
        <div>
          <a className="action-button animate blue" onClick={this.displayCards}>
            Gimme a Card!
          </a>
        </div>
        <ul>{cards}</ul>
      </div>
    );
  }
}

export default App;
