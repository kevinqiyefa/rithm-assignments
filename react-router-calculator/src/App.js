import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import './App.css';

const Operations = ({ match }) => {
  let result;
  if (match.params.operator === 'add') {
    result = Number(match.params.num1) + Number(match.params.num2);
  } else if (match.params.operator === 'subtract') {
    result = Number(match.params.num1) - Number(match.params.num2);
  } else if (match.params.operator === 'multiply') {
    result = Number(match.params.num1) * Number(match.params.num2);
  } else if (match.params.operator === 'divide') {
    result = Number(match.params.num1) / Number(match.params.num2);
  }

  return <h1>The result is {result} </h1>;
};

class App extends Component {
  render() {
    return (
      <div className="App">
        <Route exact path="/:operator/:num1/:num2" component={Operations} />
      </div>
    );
  }
}

export default App;
