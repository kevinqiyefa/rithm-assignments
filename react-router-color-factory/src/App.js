import React, { Component } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import './App.css';
import Home from './Home';
import New from './New';
import Color from './Color';

class App extends Component {
  state = {
    colors: [
      { color: 'blue', code: '#0000FF' },
      { color: 'red', code: '#FF0000' },
      { color: 'yellow', code: '#FFFF00' }
    ]
  };

  addColor = (cl, cd) => {
    this.setState({ colors: [{ color: cl, code: cd }, ...this.state.colors] });
  };
  render() {
    const color_list = this.state.colors.map((c, idx) => (
      <Route
        key={idx}
        exact
        path={`/colors/${c.color}`}
        component={() => <Color cl={c.color} cd={c.code} />}
      />
    ));
    return (
      <div className="App">
        <Switch>
          <Route
            exact
            path="/colors"
            component={() => <Home cls={this.state.colors} />}
          />
          <Route
            exact
            path="/colors/new"
            component={props => <New {...props} addCl={this.addColor} />}
          />
          {color_list}

          <Redirect to="/colors" />
        </Switch>
      </div>
    );
  }
}

export default App;
