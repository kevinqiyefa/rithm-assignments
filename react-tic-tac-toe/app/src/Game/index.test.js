import React from 'react'; // for JSX
import { shallow, mount } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import App from '.'; // import the component itself

describe('<App />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = shallow(<App />);
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});

describe('<App />', () => {
  it('ends after 8 turns', () => {
    let wrapper = mount(<App />);
    // first assert that the game isnt over
    expect(wrapper.state().over).toBe(false);
    // then set the state to just BEFORE the bug happens
    wrapper.setState({
      board: [['X', 'X', 'O'], ['O', 'X', 'O'], ['X', null, null]],
      currentPlayer: 'O',
      over: false,
      turn: 8,
      winner: null
    });
    // then select a square
    let sq21 = wrapper.find('#sq21');

    // click on the square
    sq21.simulate('click');

    // assert that the game should still not be over
    expect(wrapper.state().over).toBe(false);
  });

  it('ends after 9 turns', () => {
    let wrapper = mount(<App />);
    // first assert that the game isnt over
    expect(wrapper.state().over).toBe(false);
    // then set the state to just BEFORE the bug happens
    wrapper.setState({
      board: [['X', 'X', 'O'], ['O', 'X', 'X'], [null, 'O', 'O']],
      currentPlayer: 'X',
      over: false,
      turn: 9,
      winner: 'null'
    });

    // then select a square
    let sq20 = wrapper.find('#sq20');

    // click on the square
    sq20.simulate('click');
    // assert that the game should still not be over
    expect(wrapper.state().over).toBe(true);
    let reset = wrapper.find('#resetButton');
    reset.simulate('click');
    expect(wrapper.state().turn).toBe(1);
  });

  it('X should win', () => {
    let wrapper = mount(<App />);
    // first assert that the game isnt over
    expect(wrapper.state().over).toBe(false);
    // then set the state to just BEFORE the bug happens
    wrapper.setState({
      board: [['X', 'X', null], ['O', 'O', null], [null, null, null]],
      currentPlayer: 'X',
      over: false,
      turn: 5,
      winner: null
    });
    // then select a square
    let sq02 = wrapper.find('#sq02');

    // click on the square
    sq02.simulate('click');

    // assert that the game should still not be over
    expect(wrapper.state().winner).toBe('X');
  });

  it('O should win', () => {
    let wrapper = mount(<App />);
    // first assert that the game isnt over
    expect(wrapper.state().over).toBe(false);
    // then set the state to just BEFORE the bug happens
    wrapper.setState({
      board: [['X', 'X', null], ['O', 'O', null], ['X', null, null]],
      currentPlayer: 'O',
      over: false,
      turn: 6,
      winner: null
    });
    // then select a square
    let sq12 = wrapper.find('#sq12');

    // click on the square
    sq12.simulate('click');

    // assert that the game should still not be over
    expect(wrapper.state().winner).toBe('O');
  });

  it('Testing click on filled square', () => {
    let wrapper = mount(<App />);

    wrapper.setState({
      board: [['X', null, null], [null, null, null], [null, null, null]],
      currentPlayer: 'O',
      over: false,
      turn: 2,
      winner: null
    });
    // then select a square
    let sq00 = wrapper.find('#sq00');

    // click on the square
    sq00.simulate('click');

    // assert that the game should still not be over
    expect(wrapper.state().board[0][0]).toBe('X');
  });
});
