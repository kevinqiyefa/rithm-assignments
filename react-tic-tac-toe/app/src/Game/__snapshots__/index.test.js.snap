// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  id="board"
>
  <Board
    board={
      Array [
        Array [
          null,
          null,
          null,
        ],
        Array [
          null,
          null,
          null,
        ],
        Array [
          null,
          null,
          null,
        ],
      ]
    }
    frozen={false}
    takeTurn={[Function]}
  />
</div>
`;
